src/bezier.typ

```diff
@@ -236,55 +236,126 @@
           (right.at(0), right.at(3), right.at(1), right.at(2)))
 }
 
-/// Shorten curve by length d. A negative length shortens from the end.
+/// Approximate cubic curve length
+/// - s  (vector): Curve start
+/// - e  (vector): Curve end
+/// - c1 (vector): Control point 1
+/// - c2 (vector): Control point 2
+/// -> float Arc length
+#let cubic-arclen(s, e, c1, c2, samples: 10) = {
+  let d = 0
+  let last = none
+  for t in range(0, samples + 1) {
+    let pt = cubic-point(s, e, c1, c2, t / samples)
+    if last != none {
+      d += vector.dist(last, pt)
+    }
+    last = pt
+  }
+  return d
+}
+
+/// Shorten curve by offsetting s and c1 or e and c2
+/// by distance d. If d is positive the curve gets shortened
+/// from s on, if d is negative it gets shortened form its
+/// end e.
```

Comment:
"Shorten *the* curve"
I still don't think this is very clear, if d is positive does s stay the same and e gets closer to s? in which direction does it go as well, does e go along the curve or directly towards s?

Reply:
It goes along the curve. I'll improve the comment.

Reply:
I've updated the comment. The "linear" function does not move s/e on the curve, the non-linear function keeps the curvature by moving both points on the curve.